logic/src/commonMain/kotlin/com/wire/kalium/logic/configuration/SelfDeletingMessagesStatus.kt

```diff
@@ -17,4 +17,9 @@
  */
 package com.wire.kalium.logic.configuration
 
-data class SelfDeletingMessagesStatus(val isEnabled: Boolean, val isStatusChanged: Boolean?, val enforcedTimeoutInSeconds: Int?)
+data class SelfDeletingMessagesStatus(
+    val isFeatureEnabled: Boolean,
+    val hasFeatureChanged: Boolean?,
+    val globalSelfDeletionDuration: Int? = null,
+    val isEnforced: Boolean = false
+)
```

Comment:
I am abit confused why do we need this info to have inside kalium if hasFeatureChanged, what is this even doing ?

Reply:
This is in charge to inform AR when a feature flag setting has changed from the backend/team settings. That way, the app can display a dialog to the user, similar to how we do it for other features like file sharing, guest links and such.

Reply:
thanks for clarying, my guts telling me this is a bit weird, but perhaps it is because I have not worked with it